Source/WebCore/platform/gstreamer/PlatformSpeechSynthesizerGStreamer.cpp

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright (C) 2023 Igalia S.L. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "PlatformSpeechSynthesizer.h"
+
+#if ENABLE(SPEECH_SYNTHESIS) && USE(GSTREAMER)
+
+#include "GStreamerCommon.h"
+#include "PlatformSpeechSynthesisUtterance.h"
+#include "PlatformSpeechSynthesisVoice.h"
+#include "WebKitAudioSinkGStreamer.h"
+#include "WebKitFliteSourceGStreamer.h"
+#include <wtf/text/StringConcatenateNumbers.h>
+
+namespace WebCore {
+
+class GstSpeechSynthesisWrapper {
+    WTF_MAKE_FAST_ALLOCATED;
+    WTF_MAKE_NONCOPYABLE(GstSpeechSynthesisWrapper);
+public:
+    explicit GstSpeechSynthesisWrapper(Ref<PlatformSpeechSynthesizer>);
+    ~GstSpeechSynthesisWrapper();
+
+    void pause();
+    void resume();
+    void speakUtterance(RefPtr<PlatformSpeechSynthesisUtterance>&&);
+    void cancel();
+    void resetState();
+
+private:
+    bool handleMessage(GstMessage*);
+
+    RefPtr<PlatformSpeechSynthesisUtterance> m_utterance;
+    Ref<PlatformSpeechSynthesizer> m_platformSynthesizer;
+    GRefPtr<GstElement> m_pipeline;
+    GRefPtr<GstElement> m_src;
+};
+
+GstSpeechSynthesisWrapper::GstSpeechSynthesisWrapper(Ref<PlatformSpeechSynthesizer> synthesizer)
```

Comment:
Should probably use `Ref<PlatformSpeechSynthesizer>&& synthesizer`?

Reply:
Maybe! @shivamidow can you follow-up?

Reply:
Yes, I will.